File: 백준/Bronze/10798. 세로읽기/세로읽기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    private static final BufferedReader br = new BufferedReader(
        new InputStreamReader(System.in)
    );

    private static StringBuilder sb = new StringBuilder();
    
    public static void main(String[] args) throws IOException {
        char[][] array = new char[5][15];
        
        for (int i = 0; i < 5; i++) {
            char[] input = br.readLine().toCharArray();
            for (int j = 0; j < input.length; j++)
            array[i][j] = input[j];
        }

        for (int i = 0; i < 15; i++)    {
            for (int j = 0; j < 5; j++) {
                if (array[j][i] == 0)   continue;
                sb.append(array[j][i]);
            }
        }

        System.out.print(sb.toString());
    }
}

File: 백준/Bronze/10798. 세로읽기/세로읽기.kt
import java.io.BufferedReader
import java.io.InputStreamReader


fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    val sb = StringBuilder()

    val array = Array(5) { CharArray(15) }

    for (i in 0..4) {
        val input = readLine().toCharArray()
        for (j in input.indices) {
            array[i][j] = input[j]
        }
    }

    for (i in 0..14) {
        for (j in 0..4) {
            if (array[j][i].code == 0) {
                continue
            }
            sb.append(array[j][i])
        }
    }

    print(sb.toString())
}